docs/struct_constructors.md

```diff
@@ -0,0 +1,118 @@
+Struct constructors
+===================
+
+This documents various types of constructors that can be defined for structs in LIME IDL.
+
+Implicit constructors
+---------------------
+
+When no constructors are explicitly defined for a LIME IDL struct, Gluecodium still adds some constructors to the
+generated code. Each such struct would have at least the constructor that provides values to all uninitialized fields 
+(i.e. fields which do not have a default value specified in the LIME definition). For example,
+```
+struct MyStruct {
+    text: String
+    isMultiline: Boolean = true
+}
+```
+will have a `MyStruct(text)` constructor generated in all output languages.
+
+In addition, if the struct is marked as `@Immutable`, an all-fields constructor will be generated too. For example,
+```
+@Immutable
+struct MyStruct {
+    text: String
+    isMultiline: Boolean = true
+}
+```
+will have both `MyStruct(text)` and `MyStruct(text, isMultiline)` constructors generated.
+
+Implicit constructors are sufficient for most use cases. When the API semantics require a different set of constructors,
+or when constructors need to have logic beyond just field initialization, explicit constructor definitions could be used
+instead, as outlined in the following sections.
+
+Field constructors
```

Comment:
Is definition of field and/or custom constructor preserves implicit ones?

Reply:
Updated